### ISSUE TITLE
make not fully-connected networks in Kera

### ISSUE THREAD

**Issue by sercretalphabets** [fchollet/keras#3919]
Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by sercretalphabets** [comment]
How to make a not fully connected graph in Keras? I am trying to make a network with some nodes in hidden layer that are not connected to the input layer but are connected to the output layer. Is there any way to do this easily in Keras?
Thanks!

**Comment by hypernote** [comment]
I guess you can use a **Input** layer from [Keras Functional API](https://keras.io/getting-started/functional-api-guide/) and divide it between your hidden layer and your output. Give a look to **Multi-input and multi-output models** example.

**Comment by sercretalphabets** [comment]
Thanks for your reply. It looks like that I have to make two models and merge them somehow. Problem is that I cannot figure out how to do this.

**Comment by farizrahman4u** [comment]
Something like
```python
input = Input((10,))
a, b = Lambda(lambda x: [x[:, :10], x[:, 10:]], output_shape=lambda _:[(None, 10), (None, 10)])(input)
c = Dense(10)(a)
output =merge([c, b], mode='concat')
model = Model(input, output)
```

**Comment by sercretalphabets** [comment]
When I try this, I get the following exception:
Exception: The `get_output_shape_for` method of layer "lambda_1"" should return one shape tuple per output tensor of the layer. Found: [((None, 10), (None, 10))]

**Comment by farizrahman4u** [comment]
I have updated the code

**Comment by sercretalphabets** [comment]
Thanks a lot!

**Comment by sercretalphabets** [comment]
Thanks to your code I did this in this way, Is this correct?

X = dataset[:,0:8] #input data (768,8)
X1 = dataset[:,0:5] # part of input (768,5)
X2 = dataset[:,5:8] # part of input (768,3)
labels = dataset[:,8] #labels (768,1)

input1 = Input((5,)) #batches of 5-dimensional vectors
hideen_layer = Dense(4,activation='relu')(input1) # make a connected layer with 4 nodes on input1
input2 = Input((3,))#batches of 3-dimensional vectors
x =merge([hideen_layer, input2], mode='concat')
output = Dense(1, activation='softmax')(x)
model = Model([input1,input2], output=[output])
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit([X1,X2], labels, nb_epoch=150, batch_size=10)
